sockeye/encoder.py

```diff
@@ -932,6 +933,7 @@ def encode(self,
         :param seq_len: Maximum sequence length.
         :return: Encoded versions of input data data, data_length, seq_len.
         """
+        data = mx.sym.cast(data=data, dtype=self.dtype)
```

Comment:
as said in my previous comment, this code is also used in training and without the override-dtype flag, so we should make sure that (no-op) casting does not incur any performance penalties.

Reply:
I will try to double-ckeck that. Will be something to report as a bug in MxNet if no-op casting will have penalties.